smart_open/smart_open_lib.py

```diff
@@ -635,12 +635,26 @@ def compression_wrapper(file_obj, filename, mode):
         return file_obj
 
 
-def file_smart_open(fname, mode='rb'):
+def file_smart_open(fname, mode='rb', parents=False):
     """
     Stream from/to local filesystem, transparently (de)compressing gzip and bz2
     files if necessary.
 
     """
+    if parents:
+        from ntpath import split
+        import errno
```

Comment:
Also, why you import this two libs here? It's built-in libs, why not import them at the top of the file?